### PR TITLE
Break the cmd executor service from Hypercube out

### PR DESCRIPTION
**Pull request by jonathangreen**
## GitHub Issue

https://github.com/Islandora-CLAW/CLAW/issues/577

## What does this Pull Request do?

Break out the Service for running shell commands from Hypercube, so it can also be used in the upcoming image service.

# How should this be tested?

This one is a bit tough to test on its own. It should be tested with the upcoming Hypercube pull that changes Hypercube to use this class, to make sure Hypercube still works.

# Interested parties
@Islandora-CLAW/committers @dannylamb

### CODE REVIEW

File: src/CmdExecuteService.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+     *
+     * @param $cmd
+     * @param $data
+     *
+     * @throws \RuntimeException
+     *
+     * @return \Closure
+     *   Closure that streams the output of the command.
+     */
+    public function execute($cmd, $data)
```
Comment (jonathangreen) [inline src/CmdExecuteService.php]: I pushed the arguments up to the parent class, so this one could be used more generically. You just pass a full command to run here. 

You can see how this is used in Hypercube here:
https://github.com/Islandora-CLAW/Crayfish/pull/19/commits/fadd616b35cf577f494e1165033f58609a0bab5a

File: composer.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
     "type": "library",
     "require": {
         "islandora/chullo": "^0.0.8",
-        "symfony/http-foundation": "3.2.6"
+        "symfony/http-foundation": "3.2.6",
+        "psr/log": "^1.0.1"
```
Comment (dannylamb) [inline composer.json]: I think if we use the monolog provider from Silex then we get automatic logging of requests and uncaught errors, though I've never tried.  Definitely not a blocker, but something to try out later maybe.

**Comment by dannylamb** [comment]
Looks great.  I'm good with this.

**Comment by dannylamb** [comment]
I'll check out https://github.com/Islandora-CLAW/Crayfish/pull/19 now

File: composer.json (hunk 2 above)
Comment (jonathangreen) [inline composer.json]: I actually am using the monolog provider, but for the tests I wanted to use the generic test interfaces, so I needed that dependency.

**Pull request merged**